ospf6d/ospf6_abr.c

```diff
@@ -816,8 +816,10 @@ void ospf6_abr_old_path_update(struct ospf6_route *old_route,
 
 	for (ALL_LIST_ELEMENTS(old_route->paths, anode, anext, o_path)) {
 		if (o_path->area_id != route->path.area_id ||
-		    (memcmp(&(o_path)->origin, &(route)->path.origin,
-			    sizeof(struct ospf6_ls_origin)) != 0))
+		    ((o_path->origin.type != route->path.origin.type) ||
```

Comment:
What actual problem have you encountered here?  We XCALLOC all memory and routinely do memcmp.  How is this better?

Reply:
The standard doesn't ensure that the padding bytes are copied when the struct is passed by value, or copied in an assignment. Even if the current code doesn't pass `ospf6_ls_origin` as a value or copy it anywhere, this change will prevent any possible bugs in the future.

Reply:
Yeah, this is one of those sad cases where the code is "right" but not "_technically right_"… the padding bytes should in normal cases be zero but there's no guarantee for that.

Reply:
Someone then needs to go through all the other daemons and fix this too

Reply:
Yep I already added this to my todo list :)